pkg/integrations/v2/eventhandler/eventhandler.go

```diff
@@ -0,0 +1,437 @@
+	}
+
+	if event.ReportingInstance != "" {
+		msg.WriteString(fmt.Sprintf("reportinginstance=%s ", event.ReportingInstance))
+	}
+	if event.ReportingController != "" {
+		msg.WriteString(fmt.Sprintf("reportingcontroller=%s ", event.ReportingController))
+	}
+	if event.Source.Component != "" {
+		msg.WriteString(fmt.Sprintf("sourcecomponent=%s ", event.Source.Component))
+	}
+	if event.Source.Host != "" {
+		msg.WriteString(fmt.Sprintf("sourcehost=%s ", event.Source.Host))
+	}
+	if event.Reason != "" {
+		msg.WriteString(fmt.Sprintf("reason=%s ", event.Reason))
+	}
+	if event.Type != "" {
+		msg.WriteString(fmt.Sprintf("type=%s ", event.Type))
+	}
+	if event.Count != 0 {
+		msg.WriteString(fmt.Sprintf("count=%d ", event.Count))
+	}
+
+	msg.WriteString(fmt.Sprintf("msg=%q", event.Message))
+
+	return labels, msg.String(), nil
+}
+
+func getTimestamp(event *v1.Event) time.Time {
+	if !event.LastTimestamp.IsZero() {
+		return event.LastTimestamp.Time
+	}
+	return event.EventTime.Time
+}
+
+func newEntry(msg string, ts time.Time, labels model.LabelSet) api.Entry {
+	entry := logproto.Entry{Timestamp: ts, Line: msg}
+	return api.Entry{Labels: labels, Entry: entry}
+}
+
+// maintain "last event" state
+func (eh *EventHandler) updateLastEvent(e *v1.Event, eventTs time.Time) error {
+	eh.Lock()
+	defer eh.Unlock()
+
+	eventRv := e.ResourceVersion
+
+	if eh.LastEvent == nil {
+		// startup
+		eh.LastEvent = &ShippedEvents{Timestamp: eventTs, RvMap: make(map[string]struct{})}
+		eh.LastEvent.RvMap[eventRv] = struct{}{}
+		return nil
+	}
+
+	// if timestamp is the same, add to map
+	if eh.LastEvent != nil && eventTs.Equal(eh.LastEvent.Timestamp) {
+		eh.LastEvent.RvMap[eventRv] = struct{}{}
+		return nil
+	}
+
+	// if timestamp is different, create a new ShippedEvents struct
+	eh.LastEvent = &ShippedEvents{Timestamp: eventTs, RvMap: make(map[string]struct{})}
+	eh.LastEvent.RvMap[eventRv] = struct{}{}
+	return nil
+}
+
+func (eh *EventHandler) writeOutLastEvent() error {
+	level.Info(eh.Log).Log("msg", "Flushing last event to disk")
+
+	eh.Lock()
+	defer eh.Unlock()
+
+	if eh.LastEvent == nil {
+		level.Info(eh.Log).Log("msg", "No last event to flush, returning")
+		return nil
+	}
+
+	temp := eh.CachePath + "-new"
+	buf, err := json.Marshal(&eh.LastEvent)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(temp, buf, os.FileMode(cacheFileMode))
+	if err != nil {
+		return err
+	}
+
+	if err = os.Rename(temp, eh.CachePath); err != nil {
+		return err
+	}
+	level.Info(eh.Log).Log("msg", "Flushed last event to disk")
+	return nil
+}
+
+// RunIntegration runs the eventhandler integration
+func (eh *EventHandler) RunIntegration(ctx context.Context) error {
+	var wg sync.WaitGroup
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Quick check to make sure logs instance exists
+	if i := eh.LogsClient.Instance(eh.LogsInstance); i == nil {
+		level.Error(eh.Log).Log("msg", "Logs instance not configured", "instance", eh.LogsInstance)
+	}
```

Comment:
Is there much point in continuing here? It seems we'll have a panic in `handleEvent` when we try to send the entry later if this is nil.

Reply:
Missed this, good catch!